src/utils.ts

```diff
@@ -1157,6 +1158,38 @@ interface RequestOptions {
   headers?: http.OutgoingHttpHeaders;
 }
 
+/**
+ * Perform a get request that returns status and body.
+ * @internal
+ */
+export function get(
+  url: URL | string,
+  options: http.RequestOptions = {}
+): Promise<{ body: string; status: number | undefined }> {
+  return new Promise((resolve, reject) => {
+    /* eslint-disable prefer-const */
+    let timeoutId: NodeJS.Timeout;
+    const request = http
+      .get(url, options, response => {
+        response.setEncoding('utf8');
+        let body = '';
+        response.on('data', chunk => (body += chunk));
+        response.on('end', () => {
+          clearTimeout(timeoutId);
+          resolve({ status: response.statusCode, body });
+        });
+      })
+      .on('error', error => {
+        clearTimeout(timeoutId);
+        reject(error);
+      })
+      .end();
+    timeoutId = setTimeout(() => {
+      request.destroy(new MongoNetworkTimeoutError(`request timed out after 10 seconds`));
+    }, 10000);
+  });
+}
```

Comment:
Probably a dumb question, but this seems to duplicate a lot from the `request()` function below, should this be re-using it?

If not, I'd consider making the error message thrown here a bit more verbose and contain the URL, similar to `request()` below

Reply:
Note: we're using a different endpoint for GCP: `computeMetadata/v1/instance/service-accounts/default/identity` vs `computeMetadata/v1/instance/service-accounts/default/token`.

Reply:
Our `get` and `request` are HTTP utilities, but Steve's comment is definitely relevant for https://github.com/mongodb/node-mongodb-native/pull/3912#discussion_r1606986443